presto-tests/src/main/java/com/facebook/presto/tests/AbstractTestQueries.java

```diff
@@ -2314,6 +2314,22 @@ public void testNonEqualityJoinWithTryInFilter()
     }
 
     @Test
+    public void testTryPushedDownLeftJoinClause()
+            throws Exception
+    {
+        // this pushdowns TRY function to the sub filters
```

Comment:
not actually due to outer join. Here's the distributed query plan of this case. Please watch the `Fragment 5`
```
 Fragment 0 [SINGLE]
     Output layout: [regionkey, expr_7, expr_18]
     Output partitioning: SINGLE []
     - Output[regionkey, id, id] => [regionkey:bigint, expr_7:bigint, expr_18:bigint]
             id := expr_7
             id := expr_18
         - RemoteSource[1] => [expr_7:bigint, regionkey:bigint, expr_18:bigint]

 Fragment 1 [HASH]
     Output layout: [expr_7, regionkey, expr_18]
     Output partitioning: SINGLE []
     - Project => [expr_7:bigint, regionkey:bigint, expr_42:bigint, expr_18:bigint, expr_43:bigint, $hashvalue_61:bigint]
         - LeftJoin[("expr_42" = "expr_43")] => [expr_7:bigint, regionkey:bigint, expr_42:bigint, $hashvalue:bigint, expr_18:bigint, expr_43:bigint, $hashvalue_61:bigint]
             - Project => [expr_7:bigint, regionkey:bigint, expr_42:bigint, $hashvalue:bigint]
                 - RemoteSource[2] => [expr_7:bigint, regionkey:bigint, expr_42:bigint, $hashvalue:bigint, $hashvalue_53:bigint]
             - Project => [expr_18:bigint, expr_43:bigint, $hashvalue_61:bigint]
                 - RemoteSource[5] => [expr_18:bigint, expr_43:bigint, $hashvalue_61:bigint, $hashvalue_62:bigint]

 Fragment 2 [HASH]
     Output layout: [expr_7, regionkey, expr_7, $hashvalue_60, $hashvalue_60]
     Output partitioning: HASH [expr_7]
     - Project => [expr_7:bigint, regionkey:bigint, $hashvalue_60:bigint]
             $hashvalue_60 := "combine_hash"(BIGINT '0', COALESCE("$operator$hash_code"("expr_7"), 0))
         - LeftJoin[("regionkey_40" = "expr_41")] => [regionkey:bigint, regionkey_40:bigint, $hashvalue_54:bigint, expr_7:bigint, expr_41:bigint, $hashvalue_57:bigint]
             - Project => [regionkey:bigint, regionkey_40:bigint, $hashvalue_54:bigint]
                 - RemoteSource[3] => [regionkey:bigint, regionkey_40:bigint, $hashvalue_54:bigint, $hashvalue_55:bigint]
             - Project => [expr_7:bigint, expr_41:bigint, $hashvalue_57:bigint]
                 - RemoteSource[4] => [expr_7:bigint, expr_41:bigint, $hashvalue_57:bigint, $hashvalue_58:bigint]

 Fragment 3 [SOURCE]
     Output layout: [regionkey, regionkey, $hashvalue_56, $hashvalue_56]
     Output partitioning: HASH [regionkey]
     - Project => [regionkey:bigint, $hashvalue_56:bigint]
             $hashvalue_56 := "combine_hash"(BIGINT '0', COALESCE("$operator$hash_code"("regionkey"), 0))
         - Filter[("regionkey" < BIGINT '1')] => [regionkey:bigint]
             - TableScan[tpch:tpch:region:sf1.0, originalConstraint = ("regionkey" < BIGINT '1')] => [regionkey:bigint]
                     regionkey := tpch:regionkey

 Fragment 4 [SOURCE]
     Output layout: [expr_7, expr_7, $hashvalue_59, $hashvalue_59]
     Output partitioning: HASH [expr_7]
     - Project => [expr_7:bigint, $hashvalue_59:bigint]
             $hashvalue_59 := "combine_hash"(BIGINT '0', COALESCE("$operator$hash_code"("expr_7"), 0))
         - Project => [expr_7:bigint]
                 expr_7 := TRY("from_base"(CAST("name_3" AS VARCHAR), BIGINT '36'))
             - Filter[((NOT ("name_3" IS NULL)) AND (TRY("from_base"(CAST("name_3" AS VARCHAR), BIGINT '36')) < BIGINT '1'))] => [name_3:varchar(25)]
                 - TableScan[tpch:tpch:nation:sf1.0, originalConstraint = ((NOT ("name_3" IS NULL)) AND (TRY("from_base"(CAST("name_3" AS VARCHAR), BIGINT '36')) < BIGINT '1'))] => [name_3:varchar(25)]
                         name_3 := tpch:name

 Fragment 5 [SOURCE]
     Output layout: [expr_18, expr_18, $hashvalue_63, $hashvalue_63]
     Output partitioning: HASH [expr_18]
     - Project => [expr_18:bigint, $hashvalue_63:bigint]
             $hashvalue_63 := "combine_hash"(BIGINT '0', COALESCE("$operator$hash_code"("expr_18"), 0))
         - Project => [expr_18:bigint]
                 expr_18 := TRY("from_base"(CAST("name_14" AS VARCHAR), BIGINT '36'))
             - Filter[((NOT ("name_14" IS NULL)) AND ((TRY("from_base"(CAST("name_14" AS VARCHAR), BIGINT '36')) < BIGINT '1') OR (TRY("from_base"(CAST("name_14" AS VARCHAR), BIGINT '36')) IS NULL)))] => [name_14:varchar(25)]
                 - TableScan[tpch:tpch:nation:sf1.0, originalConstraint = (NOT ("name_14" IS NULL))] => [name_14:varchar(25)]
                         name_14 := tpch:name
```